api/src/api/post/post.services.ts

```diff
@@ -0,0 +1,98 @@
+import { prismaClient } from "utils/db";
+
+const createPost = async (authorId: string, title: string, content: string, url?: string) => {
+  if (url) {
+    const newPost = await prismaClient.post.create({
+      data: {
+        title,
+        content,
+        authorId,
+        postPhotos: {
+          create: { url: url },
+        },
+      },
+    });
+    return newPost;
+  } else {
+    const newPost = await prismaClient.post.create({
+      data: {
+        title,
+        content,
+        authorId,
+      },
+    });
+    return newPost;
+  }
+};
+const updatePost = async (postId: string, title: string, content: string, url?: string) => {
```

Comment:
Again use an images array instead of just url. The image object should looks like this `{ id?: string, url: string }`

The backend can interprete the images array as follow: 
- No id: A new image is attached to the post
- With id: An old image (which we might update if the url changed)

This also matches with `createImage` above.